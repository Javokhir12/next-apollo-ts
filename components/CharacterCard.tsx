import { Box, Button } from '@chakra-ui/react';
import Image from 'next/image';
import Link from 'next/link';

export interface CharacterCardProps {
  name: string;
  src: string;
  id: string;
}

function CharacterCard({ name, src, id }: CharacterCardProps) {
  return (
    <Box
      position="relative"
      width={250}
      pb="0.5rem"
      borderWidth="1px"
      borderRadius="lg"
      overflow="hidden"
    >
      <Image src={src} alt={name} width={250} height={225} />

      <Box as="h3" textAlign="center" my="1rem" color="gray.700" fontSize="lg">
        <a>{name}</a>
      </Box>
      <Button mx='auto' mb="0.5rem" display='block' colorScheme="green" size="md" mt="0.5rem">
        <Link href={`/characters/${id}`} passHref>
          <a>Read more</a>
        </Link>
      </Button>
    </Box>
  );
}

export default CharacterCard;
