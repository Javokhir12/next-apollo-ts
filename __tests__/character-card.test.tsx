import CharacterCard from '@/components/CharacterCard';
import renderer from 'react-test-renderer';

test('CharacterCard renders correctly', () => {
  const tree = renderer
    .create(
      <CharacterCard id="1" name="Rick" src="https://test.com/rick.png" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
