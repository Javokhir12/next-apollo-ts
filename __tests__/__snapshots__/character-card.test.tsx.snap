// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard renders correctly 1`] = `
<div
  className="css-1a2x920"
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27250%27%20height=%27225%27/%3e"
        style={
          {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="Rick"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="Rick"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=640&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=640&q=75 2x"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <h3
    className="css-120stqy"
  >
    <a>
      Rick
    </a>
  </h3>
  <button
    className="chakra-button css-196aufa"
    type="button"
  >
    <a
      href="/characters/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Read more
    </a>
  </button>
</div>
`;

exports[`Link renders correctly 1`] = `
<div
  className="css-1a2x920"
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27250%27%20height=%27225%27/%3e"
        style={
          {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="Rick"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="Rick"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=640&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=640&q=75 2x"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <h3
    className="css-120stqy"
  >
    <a>
      Rick
    </a>
  </h3>
  <button
    className="chakra-button css-196aufa"
    type="button"
  >
    <a
      href="/characters/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Read more
    </a>
  </button>
</div>
`;
